weasyl/middleware.py

```diff
@@ -141,25 +139,26 @@ def status_check_tween(request):
     return status_check_tween
 
 
+def database_session_cleanup_tween_factory(handler, registry):
+    """
+    A tween that cleans up the thread-local database session after every request.
+    """
+    def database_session_cleanup_tween(request):
+        def cleanup(request):
+            d.sessionmaker.remove()
+
+        request.add_finished_callback(cleanup)
+        return handler(request)
+
+    return database_session_cleanup_tween
+
+
 # Properties and methods to enhance the pyramid `request`.
 def pg_connection_request_property(request):
     """
     Used for the reified pg_connection property on weasyl requests.
     """
-    db = d.sessionmaker()
-    try:
-        # Make sure postgres is still there before issuing any further queries.
-        db.execute('SELECT 1')
-    except OperationalError:
-        request.log_exc()
-        raise HTTPServiceUnavailable("database error")
-
-    # postgresql is still there. Register clean-up of this property.
-    def cleanup(request):
-        db.close()
-
-    request.add_finished_callback(cleanup)
-    return db
+    return d.sessionmaker()
```

Comment:
> What's the consequence of removing the checking for the database still being there?

I’m hoping that it’s “slightly improving response times”. Removing things without knowing why they were added is a little worrying, but it seems reasonable to think that the only thing `SELECT 1` can do is provide a more specific error to the user – something that might be better handled in middleware if it really turns out to be necessary.